src/displayBackend/wayland/Shell.cpp

```diff
@@ -53,21 +53,34 @@ ShellSurfacePtr Shell::createShellSurface(SurfacePtr surface)
 
 void Shell::init()
 {
-	mWlShell = bind<wl_shell*>(&wl_shell_interface);
+	mXDGShell = bind<xdg_wm_base*>(&xdg_wm_base_interface);
 
-	if (!mWlShell)
+	if (!mXDGShell)
 	{
 		throw Exception("Can't bind shell", errno);
 	}
 
+	mWlListener = { .ping = sPingHandler};
```

Comment:
AFAIK, designated initializer is available starting from c++20. It will not compiled on earlier c++ version.